### PR TITLE
[fix](cast) fix wrong result of casting empty string to array date

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jacktengg** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.94 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17162310099 Bytes

File: regression-test/suites/query_p0/sql_functions/cast_function/test_cast_string_to_array.groovy [hunk 1]
```diff
@@ -41,5 +41,14 @@ suite("test_cast_string_to_array") {
...
     qt_sql """ select cast ("[a,b,c,,,]" as array<string>) """
     qt_sql """ select cast ("[1.34,2.01,,,]" as array<decimal(10, 3)>) """
+
+    sql """ ADMIN SET FRONTEND CONFIG ("enable_date_conversion" = "false"); """
+    qt_sql """ select cast ("[2022-09-01,,]" as array<date>) """
+    qt_sql """ select cast ("[2022-09-01,,]" as array<string>) """
+    qt_sql """ select cast(cast ("[2022-09-01,,]" as array<string>) as array<date>) """
+
+    sql """ ADMIN SET FRONTEND CONFIG ("enable_date_conversion" = "true"); """
     qt_sql """ select cast ("[2022-09-01,,]" as array<date>) """
```
Comment (yiguolei) [inline regression-test/suites/query_p0/sql_functions/cast_function/test_cast_string_to_array.groovy:51]: what's the difference between [2022-09-01, , ]  and [2022-09-01, NULL, NULL]  ?

Comment (jacktengg) [reply regression-test/suites/query_p0/sql_functions/cast_function/test_cast_string_to_array.groovy:51]: We and doris cast empty string to null date:
```
mysql [(none)]>select cast('' as date);
+--------------------+
| cast('' as DATEV2) |
+--------------------+
| NULL               |
+--------------------+
1 row in set (0.01 sec)
```

The casting result `[2022-09-01, , ]` should be not correct, I'll take a look.

**Comment by hello-stephen** [comment]
run compile

**Comment by yiguolei** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.86 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17168013437 Bytes

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**